### DISCUSSION TITLE
To All  Mac User using M processor

### DISCUSSION THREAD

**Discussion by Creative-comfyUI** [comfyanonymous/ComfyUI#4187] [General]
I thought this might help users with Mac M processors with Sonoma and memory. < 32 G  

- First of all, rest assured that all IA libraries are always up to date, like Pytorch and even Python. 
- Empty the cache on your hard drive. You can gain up to 10 gigabytes on your disk depending on the type of application you are using. 
- Make sure you have maximum space on your main hard drive. Even 25 Giga is sometimes not enough if you don't have 32 G of Ram. ComfyUI uses your hard drive, not just your memory. 
- Firefox, even though it uses a lot of memory, seems to be more stable for ComfyUI than other browsers. Since Sonoma Safari seems to change the way it manages the memory. More than once it crashes the system. 
- If you are creating something very complex workflow, make sure that no other application is  running in the backend or that no other application is open. 
- If you render complex workflow  it is good times to times to quit the browser and the terminal and start again it will help you to don't have memory problem or see your system crashed which arrive more often since Sonoma.

**Comment by polarNerd204** [comment]
I posted my frustration regarding this on Reddit…with M2 Ultra with 128g ram is hardly using anything to make images or video. 70% ram and 4% ram cpu usage are I think the highest I’ve seen it go. I’m not expecting 4090 speeds but I do expect the performance to be using what it has.  I’m gonna try emptying the system cache and seeing how room I can make on the internal hd. But if you know of some other settings I could change in python or cui, or Mac OS I’d love to hear them.


This Terminal info may help:

Terminal window for comfy-

Checkpoint files will always be loaded safely.

Total VRAM 131072 MB, total RAM 131072 MB

pytorch version: 2.5.1

Set vram state to: SHARED

Device: mps

Using sub quadratic optimization for attention, if you have memory or speed issues try using: --use-split-cross-attention

[Prompt Server] web root: /Users/me/ComfyUI/web

Note: NumExpr detected 24 cores but "NUMEXPR_MAX_THREADS" not set, so enforcing safe limit of 16.

NumExpr defaulting to 16 threads.
___________________________________________

also checked activity monitor : see image

_______________

how can I alter this to get some faster generation times?

**Reply by Creative-comfyUI** [reply]
Start ComfyUI uising  this 
PYTORCH_ENABLE_MPS_FALLBACK=1 python3 main.py 

This seems to help in some cases, but even with this it can take some time for some nodes.  

The best way I have found is to start my computer using only ComfyUI. I don't open any other application except the browser. 

The other probem is don't let you computer to be in sleeping mode. This will slow down or put the rendering on pause
